Update Project Utilization by Month Report for Race & Ethnicity

dtgreiner opened the pull request:
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Adds the new Race & Ethnicity changes to the Monthyl Project Utilization report

## Type of change
[//]: # 'remove options that are not relevant'
- [X] New feature (adds functionality)

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

dtgreiner committed:
Update Project Utilization by Month Report for Race & Ethnicity

eanders commented on drivers/ma_reports/app/models/ma_reports/monthly_performance/report.rb:
```diff
@@ -18,6 +18,12 @@ class Report < SimpleReports::ReportInstance
     has_many :projects
 
+    private def expires_in
+      return 30.seconds if Rails.env.development?
```
Might want to increase this even for development since the data is pre-calculated and shouldn't be changing. (Maybe 5 days?)

eanders commented on drivers/ma_reports/app/models/ma_reports/monthly_performance/report.rb:
```diff
@@ -334,14 +338,118 @@ def demographic_breakdowns
+        white_hispanic_latinaeo: ->(race_data, bucket) { single_race_latinaeo(race_data, 'white', bucket) },
+
+        hispanic_latinaeo: ->(race_data, bucket) do
+          bucket ||= []
+          race_data.select { |enrollment| enrollment['hispanic_latinaeo'] }.
+            each do |e|
+              bucket << e['id'] if e.excluding('id').count { |m| m.count(true) == 1 } == 1
+            end
```
Let's make a method `only_latinaeo` that mirrors the behavior of `single_race` but is specific to hispanic_latinaeo[

(Same with Multi-racial and multi-with-hispanic/latin@.)

That should allow `def lookups` to be identical for this file and `drivers/system_pathways/app/models/system_pathways/equity/race_and_ethnicity.rb`.

Different methods will be defined for `only_latinaeo` etc.

eanders commented on drivers/ma_reports/app/models/ma_reports/monthly_performance/report.rb:
```diff
@@ -334,14 +338,118 @@ def demographic_breakdowns
+      race_ethnicities_breakdown
+    end
+
+    def race_col_lookup
```
Let's centralize this in HudUtility2024 and remove/replace from `drivers/system_pathways/app/models/system_pathways/report.rb` as well.

eanders commented on app/models/filters/filter_base.rb:
```diff
@@ -565,6 +565,7 @@ def apply_client_level_restrictions(scope, chronic_at_entry: true)
       scope = filter_for_age(scope)
       scope = filter_for_gender(scope)
       scope = filter_for_race(scope)
+      scope = filter_for_race_ethnicity_combinations(scope)
```
Let's not add this as a filter and just use the breakdowns.

dtgreiner committed:
QA Updates:
- Revert filter to Race instead of Race/Ethnicity
- Move Race Ethnicity Calculations to where it can be shared
- Update System Pathways report to use shared calculations

dtgreiner committed:
Update System pathways race_col_lookup to use HudUtility

dtgreiner commented:
@eanders This is ready for a second round of review.

eanders commented on app/models/concerns/race_and_ethnicity_calculations.rb:
```diff
@@ -0,0 +1,95 @@
+    end
+  end
+
+  private def lookups
```
Given that this is in a concern now, I think we should rename `lookups` to something like `race_and_ethnicity_lookups` to decrease collisions with existing methods.

eanders reviewed (approved):
@dtgreiner this is a HUGE improvement, thank you!  Only one request to rename a method.

dtgreiner committed:
method name change to prevent collisions

dtgreiner commented:
@eanders The method name change has been added to the PR. Thanks!
